### DISCUSSION TITLE
能否知道当前的内容和初始化的内容是否一样（内容被编辑了）

### DISCUSSION THREAD

**Discussion by chenxiaoleizi** [umodoc/editor#34] [Q&A] [answered]
用户离开编辑器页面时，如果内容被修改了我希望提醒用户离开页面内会放弃所作的修改，能否知道当前的内容被修改了？

**Comment by umodoc** [comment]
利用编辑器`change`事件，见：https://editor.umodoc.com/cn/docs/events
定义一个变量，默认为`false`，内容`change`以后，将该标记设为`true`，用户离开页面时，根据该标记判断是否要弹出提示。

**Reply by chenxiaoleizi** [reply]
那如果用户先编辑了，又将所有的都撤销了，这种情况怎么办？

**Reply by umodoc** [reply]
如果有更深层次的需求，就需要`diff`了，可以借助第三方的库或者自行实现。

**Reply by chenxiaoleizi** [reply]
![图片](https://github.com/user-attachments/assets/41e8fa37-144a-4d23-b6a0-31d00c564a53)
如果能拿到undo redo 的栈信息，通过判断栈的length，是否可以实现啊？

**Reply by umodoc** [reply]
编辑器的相关信息可以通过`getEditor()`获取。

**Reply by chenxiaoleizi** [reply]
我通过对比 setContent() 时的 content 和 getContent 的 content可以吗？

**Reply by umodoc** [reply, marked as answer]
可以
